### PR TITLE
Fix multiple machines issues

### PR DESCRIPTION
**Pull request by abdfnx**
this pull request is fixing these issues

* #9 
* #10

### CODE REVIEW

**Comment by abdfnx** [comment]
on the next time when you want to publish to a new release

1. navigate to `actions` tap
how
2. click `Release` action
3. press `Run workflow` and enter the new release version
4. then press <a href="https://github.com/dlvhdr/gh-prs/actions/workflows/release.yml"></a>

**Comment by abdfnx** [comment]
@dlvhdr don't merge this pull request until I finish from it 👍

**Comment by abdfnx** [comment]
@dlvhdr now you can merge

Congrats, all your issues are fixed

**Comment by abdfnx** [comment]
but first you have to create a repo secret and name it `ACCESS_TOKEN` for release workflow
because github cli needs your access token to releasing a new release

**Comment by abdfnx** [comment]
1. go to https://github.com/abdfnx/gh-prs/settings/secrets/actions
2. press `New repository secret`
3. name it `ACCESS_TOKEN`
4. and put your github access token as value

**Comment by abdfnx** [comment]
And congratulations on reaching `100` github stars

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -1,16 +1,32 @@
...
-      - uses: cli/gh-extension-precompile@v1
+
+      - name: Setup Go
+        uses: actions/setup-go@v2.1.4
+        with:
+          go-version: "1.17.3"
+
+      - name: Release
+        run: |
+          echo ${{ secrets.ACCESS_TOKEN }} | gh auth login --with-token
```
Comment (dlvhdr) [inline .github/workflows/release.yml:30]: How does the gh-extension-precompile action work without having an access token?
I did see this in their `action.yml`: `GITHUB_TOKEN: ${{ github.token }}`.
Can we use the same thing?

File: gh-prs [hunk 2]
```diff
@@ -0,0 +1,56 @@
+#!/bin/bash
+set -e
+
+repo="dlvhdr/gh-prs"
+tag="v0.1.1"
```
Comment (dlvhdr) [inline gh-prs:5]: How are we going to maintain this value?
If I understand correctly, the piece of code that uses it below is for when a user clones this repo and doesn't install the extension through the `gh` cli, right?
And if that's the case, we download the right exe from the release.

In that case, can't we download the latest release?
I checked, and by not providing a tag - `gh release download` will fetch the latest version.
So I think it's safe to remove

**Review by dlvhdr — CHANGES_REQUESTED** [review]

**Comment by dlvhdr** [comment]
This looks amazing! Thanks @abdfnx 

I have a couple of questions, since I'm not familiar with github actions...

File: .github/workflows/release.yml (hunk 1 above)
Comment (abdfnx) [reply .github/workflows/release.yml:30]: ok I will test it, but it's better to use this command `echo ${{ secrets.ACCESS_TOKEN }} | gh auth login --with-token` for more functionality

File: gh-prs (hunk 2 above)
Comment (abdfnx) [reply gh-prs:5]: I test it by install it like this

```bash
gh extension install abdfnx/gh-prs
```

and it's works correctly

**Comment by abdfnx** [comment]
> This looks amazing! Thanks @abdfnx
> 
> I have a couple of questions, since I'm not familiar with github actions... How did you test it? Did you use https://github.com/nektos/act?

I use it from a long time

File: .github/workflows/release.yml (hunk 1 above)
Comment (dlvhdr) [reply .github/workflows/release.yml:30]: See https://cli.github.com/manual/gh_help_environment

Comment (abdfnx) [reply .github/workflows/release.yml]: done, I used `${{ github.token }}` and it's worked

Comment (abdfnx) [reply .github/workflows/release.yml]: now you can merge it

**Review by abdfnx — COMMENTED** [review]
👍

**Comment by abdfnx** [comment]
go ahead @dlvhdr

File: gh-prs (hunk 2 above)
Comment (abdfnx) [reply gh-prs]: Sorry, I misspelled the tag

**Comment by abdfnx** [comment]
awesome, you have a huge growth in the number of stars

[Stargazers over time](https://starchart.cc/dlvhdr/gh-prs)

File: gh-prs [hunk 3]
```diff
@@ -0,0 +1,59 @@
+#!/bin/bash
+set -e
+
+# https://github.com/dlvhdr/gh-prs/pull/11#discussion_r756341767
+LATEST_VERSION=$(curl --silent "https://api.github.com/repos/dlvhdr/gh-prs/releases/latest" | grep '"tag_name":' | sed -E 's/.*"([^"]+)".*/\1/')
```
Comment (dlvhdr) [inline gh-prs:5]: I think you can just omit the tag from the `gh release download` command.
Having a curl request at the top of the script will slow it down.

File: gh-prs [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+  exe="cmd.out"
+
+  cd "${extensionPath}" > /dev/null
+  go build -o "${exe}"
+  cd - > /dev/null
+else
+  if [[ ! -x "${extensionPath}/bin/${exe}" ]]; then
+    mkdir -p bin
+    rm -f "${extensionPath}/bin/*"
+    gh release -R"${repo}" download "${tag}" -p "${exe}" --dir="${extensionPath}/bin"
```
Comment (dlvhdr) [inline gh-prs:54]:
```suggestion
    gh release -R"${repo}" download -p "${exe}" --dir="${extensionPath}/bin"
```

File: gh-prs (hunk 3 above)
Comment (abdfnx) [reply gh-prs:5]: you got it

**Comment by abdfnx** [comment]
done

**Comment by abdfnx** [comment]
done, I removed `tag` variable

File: .github/workflows/release.yml [hunk 5]
```diff
@@ -1,16 +1,32 @@
...
-      - uses: cli/gh-extension-precompile@v1
+
+      - name: Setup Go
+        uses: actions/setup-go@v2.1.4
+        with:
+          go-version: "1.17.3"
+
+      - name: Release
+        run: |
+          echo ${{ github.token }} | gh auth login --with-token
```
Comment (dlvhdr) [inline .github/workflows/release.yml:30]: I don't think you need this since `gh` should automatically pick up the GITHUB_TOKEN
```suggestion
```

**Review by dlvhdr — APPROVED** [review]
Thanks again :)

File: .github/workflows/release.yml (hunk 5 above)
Comment (abdfnx) [reply .github/workflows/release.yml:30]: 👍

Comment (abdfnx) [reply .github/workflows/release.yml]: done

Comment (abdfnx) [reply .github/workflows/release.yml]: I make `github.token` as a env variable

**Comment by abdfnx** [comment]
everything is done, anything else?

File: .github/workflows/release.yml [hunk 6]
```diff
@@ -1,16 +1,32 @@
...
-      - "v*"
+  workflow_dispatch:
+    inputs:
+      release:
+        description: "The New Release"
+        required: true
+
+env:
+  RELEASE: ${{ github.event.inputs.release }}
+  GITHUB_TOKEN: ${{ github.token }}
```
Comment (dlvhdr) [inline .github/workflows/release.yml:12]: nice :) thanks

**Review by dlvhdr — APPROVED** [review]

**Comment by abdfnx** [comment]
and now, all your issues are fixed

**Comment by abdfnx** [comment]
try to release a new version

**Comment by dlvhdr** [comment]
@abdfnx it worked but we don't publish a darwin-amd64 exe..
So need to fix that

**Comment by abdfnx** [comment]
> @abdfnx it worked but we don't publish a darwin-amd64 exe.. So need to fix that

@dlvhdr `darwin-x86_64` === `darwin-amd64`, it's same arch

**Comment by dlvhdr** [comment]
I guess that's just how the `gh extension install` command works.. It looks for an exe named `darwin-amd64`

**Comment by abdfnx** [comment]
@dlvhdr you've to merge this pull #16

**Pull request merged**